src/Classes/PopupDialog.lua

```diff
@@ -55,6 +58,9 @@ function PopupDialogClass:Draw(viewPort)
 	if self.scrollBarFunc then
 		self.scrollBarFunc()
 	end
+	if self.resizeFunc then
+		self.resizeFunc()
+	end
```

Comment:
It might be more generically useful to check the height and width compared to the viewport/screensize and if it doesn't fit anymore, resize accordingly instead of making each popup implement their own logic for it.